### ISSUE TITLE
Batch renormalization

### ISSUE THREAD

**Issue by SeanXiaoyuSun** [FluxML/Flux.jl#1312]
Hi All,

Looks that [batch renormalization](https://arxiv.org/pdf/1702.03275.pdf) (which is supported by TensorFlow) has not been implemented in Flux. Will that be of any interest to be added to the current BatchNorm layer?

**Comment by DhairyaLGandhi** [comment]
I'd like to see it in Flux. It would make sense to support a kernel and layer version. Good to support cuda as well.

**Comment by SeanXiaoyuSun** [comment]
The layer version (by modifying current `BatchNorm`) is up and running for testing now. I'm not familiar with GPU support. Are there any specific changes needed for supporting cuda?

**Comment by CarloLucibello** [comment]
gpu support should be straightforward, probably will come out of the box. You are welcome to submit a PR if you want
